src/package.cc

```diff
@@ -25,4 +25,29 @@ Package::Package() : id_(next_id_++), fill_min_(0), fill_max_(std::numeric_limit
 
 Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : name_(name), id_(next_id_++), fill_min_(fill_min), fill_max_(fill_max), strategy_(strategy) {}
 
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg) {
+  if (r->quantity() < pkg->fill_min()) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  std::vector<Resource::Ptr> rs;
+  Resource::Ptr r_pkgd;
+  double fill_mass;
+  if (pkg->strategy() == "first") {
+    fill_mass = pkg->fill_max();
```

Comment:
What if the requested quantity is more than N packages and the extra increment is less than the minimum fill?

Reply:
Right now `first` strategy will just return N full packages. When designing this I did think that maybe we want a first-relaxed strategy that then goes back and tries to remove some material (up to `max`-`min`) from packages one by one until it can fill N+1 packages, but right now that's not implemented